src/ReduxFormContext.js

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react'
+
+export const ReduxFormContext = React.createContext(null)
+
+export const withReduxForm = Component => {
+  class Hoc extends React.Component {
+    render() {
+      const { forwardedRef, ...rest } = this.props
+      return React.createElement(ReduxFormContext.Consumer, {
+        children: _reduxForm =>
+          React.createElement(Component, {
```

Comment:
@erikras 	Possible optimization: if the Component is a functional component (`typeof Component === 'function'`) this could drop the `createElement` call, just call the function directly with the props and thereby shorten the React tree by 1 component?